src/bindings/swig/ruby/tests/testruby_keyset.rb

```diff
@@ -48,4 +48,102 @@ def test_keySet_append
 
     end  
   end
+
+
+  def test_keySet_each_enumeralbe
+    assert_nothing_raised do
+      a = Array.new
+      # create test keys
+      a << Kdb::Key.new("user/k0", value: "v0", owner: "me")
+      a << Kdb::Key.new("user/k1", value: "v1", owner: "you")
+      a << Kdb::Key.new("user/k2", value: "v2", owner: "john", m2: "a")
+      a << Kdb::Key.new("user/k3", value: "v3", owner: "jane")
+      a << Kdb::Key.new("user/k4", value: "v4", owner: "max")
+      a << Kdb::Key.new("user/k5", value: "v5", owner: "bob", m2: "b")
+      a << Kdb::Key.new("user/k6", value: "v6", owner: "alice")
+      a << Kdb::Key.new("user/k7", value: "v7", owner: "fritz", m3: "c")
+      a << Kdb::Key.new("user/k8", value: "v8", owner: "anton")
+      a << Kdb::Key.new("user/k9", value: "v9", owner: "berta")
+
+      ks = Kdb::KeySet.new
+
+      # populate keySet
+      a.each { |e| ks << e }
```

Comment:
> Array 'a' was just used to

Yes, I meant that something like this is needed often, so it is nice if it is easy to transform build-in data structures to KeySet and the other way round.

Similar unit tests exist for C++, too. There is also a unit-test that appends key in every possible order and checks if all of those keysets behave identical. It is not necessary to do this in ruby, too, but every constructor should be tested.

> `ks = Kdb::KeySet.new [Kdb::Key.new("user/k1"), Kdb::Key.new("user/k2")]`

Yes, that would be nice, too!